core/touch.js

```diff
@@ -0,0 +1,273 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Touch handling for Blockly.
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+goog.provide('Blockly.Touch');
+
+goog.require('goog.events');
+goog.require('goog.events.BrowserFeature');
+
+/**
+ * Which touch events are we currently paying attention to?
```

Comment:
We are paying attention to multiple events in series, but only ones with this identifier.  So we'd listen to a touchstart and a lot of touchmoves and then a touchend, for instance.